model/task/task.go

```diff
@@ -2486,82 +2487,89 @@ func (t *Task) Insert() error {
 // are also archived.
 func (t *Task) Archive() error {
 	if t.DisplayOnly && len(t.ExecutionTasks) > 0 {
-		execTasks, err := FindAll(db.Query(ByIds(t.ExecutionTasks)))
+		err := ArchiveMany([]Task{*t})
+		return errors.Wrapf(err, "archiving display task '%s'", t.Id)
+	} else {
+		// Archiving a single task.
+		archiveTask := t.makeArchivedTask()
+		err := db.Insert(OldCollection, archiveTask)
 		if err != nil {
-			return errors.Wrap(err, "retrieving execution tasks")
+			grip.Error(message.WrapError(err, message.Fields{
+				"archive_task_id": archiveTask.Id,
+				"old_task_id":     archiveTask.OldTaskId,
+				"execution":       t.Execution,
+				"display_only":    t.DisplayOnly,
+			}))
+			return errors.Wrap(err, "inserting archived task into old tasks")
 		}
-		if err = ArchiveMany(execTasks); err != nil {
-			return errors.Wrap(err, "archiving execution tasks")
+		err = UpdateOne(
+			bson.M{IdKey: t.Id},
+			bson.M{
+				"$unset": bson.M{
+					AbortedKey:              "",
+					AbortInfoKey:            "",
+					OverrideDependenciesKey: "",
+				},
+				"$set": bson.M{
+					ExecutionKey: t.Execution + 1,
+				},
+			})
+		if err != nil {
+			return errors.Wrap(err, "updating task")
 		}
+		t.Aborted = false
+		return nil
 	}
+}
 
-	archiveTask := t.makeArchivedTask()
-	err := db.Insert(OldCollection, archiveTask)
-	if err != nil {
-		grip.Error(message.WrapError(err, message.Fields{
-			"archive_task_id": archiveTask.Id,
-			"old_task_id":     archiveTask.OldTaskId,
-			"execution":       t.Execution,
-			"display_only":    t.DisplayOnly,
-		}))
-		return errors.Wrap(err, "inserting archived task into old tasks")
-	}
-	err = UpdateOne(
-		bson.M{IdKey: t.Id},
-		bson.M{
-			"$unset": bson.M{
-				AbortedKey:              "",
-				AbortInfoKey:            "",
-				OverrideDependenciesKey: "",
-			},
-			"$inc": bson.M{ExecutionKey: 1},
-		})
-	if err != nil {
-		return errors.Wrap(err, "updating task")
-	}
-	t.Aborted = false
-	return nil
+func shouldRestartTask(resetFailedWhenFinished bool, resetWhenFinished bool, status string) bool {
```

Comment:
I know this is a private function but let's still add a comment to explain the inputs since the use case is niche.

Additionally, since this is it's own function, it would be cleaner to break these into logically distinct pieces that can be commented to explain meaning. 

```
if resetWhenFinished { return true }
if resetFailed && isFailed {return true}
else return false 
```
is a clearer way of breaking this up

Reply:
Deleting the help function as mentioned in another thread!